File: Wrapping/WrapITK/Python/Tests/CastImageFilter.py
#
#  Example on the use of the CastImageFilter
#

import itk
from sys import argv
itk.auto_progress(2)

dim = 2
IType = itk.Image[itk.US, dim]
OIType = itk.Image[itk.UC, dim]

reader = itk.ImageFileReader[IType].New( FileName=argv[1] )
filter  = itk.CastImageFilter[IType, OIType].New( reader )
writer = itk.ImageFileWriter[OIType].New( filter, FileName=argv[2] )

writer.Update()

File: Wrapping/WrapITK/Java/Tests/CastImageFilter.java
/**
 *  Example on the use of the CastImageFilter
 *
 */

import InsightToolkit.*;

public class CastImageFilter
{
  public static void main( String argv[] )
  {
    System.out.println("CastImageFilter Example");

    itkImageFileReaderIUC2_Pointer reader = itkImageFileReaderIUC2.itkImageFileReaderIUC2_New();
    itkImageFileWriterIUS2_Pointer writer = itkImageFileWriterIUS2.itkImageFileWriterIUS2_New();

    itkCastImageFilterIUC2IUS2_Pointer filter = itkCastImageFilterIUC2IUS2.itkCastImageFilterIUC2IUS2_New();

    filter.SetInput( reader.GetOutput() );
    writer.SetInput( filter.GetOutput() );

    reader.SetFileName( argv[0] );
    writer.SetFileName( argv[1] );

    writer.Update();
  }

}


